core/src/main/java/com/yahoo/oak/EntryArray.java

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright 2020, Verizon Media.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.yahoo.oak;
+
+import sun.misc.Unsafe;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/*
+* EntryArray is array of entries based on primitive array of longs.
+* Entry must include at least Key Reference and Value Reference fields, but might include more.
+* */
```

Comment:
Move the field chart from EntrySet to here and describe the optional additional fields

Reply:
done